.github/workflows/ui-test.yml

```diff
@@ -0,0 +1,72 @@
+# Run the UI tests
+
+name: UI Tests
+
+on:
+  push:
+    branches: [ main ]
+  # We are not running on Pull Request, since we need the Google API Key
```

Comment:
And why again do branches have access to that key, but not PRs? Someone could make a random PR just to run something against Google with our API key and close the PR?

Reply:
Action on Commit on the repo runs after an maintainer have pushed a commit and thus github consider this as safe and export the "secrets" to the github action environnement. 
THe same would be true if a PR was made from a branch in the local repo. 
But when we can a PR run from branch from random people on the internet (from various forks), github to not expose the secrets as a security measure. 
Does that answer the question?

Reply:
In short, yes to your second question.